### ISSUE TITLE
`unicode_emoji` is not set to None when `icon` is passed in the role.edit()

### ISSUE THREAD

**Issue by argo0n** [Pycord-Development/pycord#874]
### Summary

When a role has a Unicode emoji set as the icon, using `role.edit(icon=bytes)` changes the icon, but from the Discord client, it appears as if nothing has changed since the role still has a Unicode emoji. This is as the `unicode_emoji` is not set to None when `icon` is passed.

### Reproduction Steps

Use `await role.edit(unicode_emoji=emoji)` to set an emoji as the icon for the role.
Use `await role.edit(icon=bytes)` to set a custom image as the icon for the role.
After completing this, the role's icon will still show the emoji as the role's icon, despite the role's `icon` attribute showing that a custom image was set successfully.

### Minimal Reproducible Code

```python
@commands.has_guild_permissions(administrator=True)
@commands.command(name='customroleicon')
async def customroleicon(self, ctx):
    pycord_emoji = await commands.PartialEmojiConverter.convert(self, ctx, '<:pycord:881477002991915048>')
    role = discord.utils.get(ctx.guild.roles, name="Minion")
    if pycord_emoji is None or role is None:
        return await ctx.send('emoji or role is none')
    emoji_bytes = await pycord_emoji.read()
    await role.edit(icon=emoji_bytes)
    await ctx.send('done')

@commands.has_guild_permissions(administrator=True)
@commands.command(name='customemojiicon')
async def customemojiicon(self, ctx):
    UnicodeEmoji = '😬'
    role = discord.utils.get(ctx.guild.roles, name="Minion")
    if UnicodeEmoji is None or role is None:
        return await ctx.send('emoji or role is none')
    await role.edit(unicode_emoji=UnicodeEmoji)
    await ctx.send('done')

@commands.has_guild_permissions(administrator=True)
@commands.command(name='roledetails')
async def roledetails(self, ctx):
    role = discord.utils.get(ctx.guild.roles, name="Minion")
    await ctx.send(f"Icon: {role.icon}\nUnicode Emoji: {role.unicode_emoji} ")
```

### Expected Results

**Step 1**
With no custom role icon **at all**, I run the command `roledetails` that returns 
>Icon: None
Unicode Emoji: None

And no role icon is displayed on the user with the role.

**Step 2**
I then run `customroleicon` to change the role's icon to the Pycord logo. **The Pycord logo is now displayed** beside the user's name, and `roledetails` shows:
>Icon: https://cdn.discordapp.com/role-icons/912396660305436684/321b18637c650c1b95b3ee6d742209ba.png?size=1024"
Unicode Emoji: None

I then run `customemojiicon` to change the role's icon to the 😬 Unicode emoji. 😬 **is now displayed** beside the user's name, and `roledetails` shows:
>Icon: None
Unicode Emoji: 😬

I expected that since the Pycord Documentation states that `If this argument (unicode_emoji) is passed, icon is set to None.`

**Step 3**
I then run `customroleicon` again to change the role's icon back to the Pycord logo. It **should display the Pycord logo** beside the user's name. Running `roledetails` should show :
>Icon: https://cdn.discordapp.com/role-icons/912396660305436684/321b18637c650c1b95b3ee6d742209ba.png?size=1024
Unicode Emoji: None

### Actual Results

At Step 3, When I run `customroleicon` again to change the role's icon back to the Pycord logo, it is _apparently_ successful, but 😬 is still shown beside the user's name. I run `roledetails`, and it shows:
>Icon: https://cdn.discordapp.com/role-icons/912396660305436684/321b18637c650c1b95b3ee6d742209ba.png?size=1024
Unicode Emoji: 😬

Because of this, the custom role icon (the Pycord logo) is not displayed. I presume it was supposed to be similar to changing `unicode_emoji`, where passing `icon` should set `unicode_emoji` to None.

### Intents

discord.Intents(guilds = True, members = True, presences = True, messages = True, reactions = True, emojis = True, invites = True, voice_states = True)

### System Information

- Python v3.8.10-final
- py-cord v2.0.0-alpha
    - py-cord pkg_resources: v2.0.0a4780+g1336473e
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.22000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context
![DiscordPTB_bcBn6u9Uwy](https://user-images.githubusercontent.com/67531837/151557978-0db31ee7-b085-469c-9629-63616d670723.png)

**Comment by argo0n** [comment]
Opened a PR for this issue at https://github.com/Pycord-Development/pycord/pull/876

**Comment by Dorukyum** [comment]
Fixed in #876.
